backend/src/utils/httpUtils.ts

```diff
@@ -89,6 +103,9 @@ export const proxyCall = (
                 data += chunk;
               })
               .on('end', () => {
+                if (rejectOnHttpErrorCode && res.statusCode >= 400) {
+                  reject(createCustomError(res.statusMessage, data, res.statusCode));
+                }
                 resolve(data);
```

Comment:
Is it possible we consider sending two bits of data back -- and you can handle 400s for Prometheus and not all proxy calls (k8s might use 400s for another use-case with status objects)